src/python_testing/TC_DEM_2_7.py

```diff
@@ -0,0 +1,324 @@
+        await self.send_request_constraint_based_forecast(constraintList, cause=Clusters.DeviceEnergyManagement.Enums.AdjustmentCauseEnum.kLocalOptimization, expected_status=Status.ConstraintError)
+
+        self.step("8")
+        now = datetime.datetime.now(tz=datetime.timezone.utc) - datetime.datetime(2000, 1, 1, 0, 0, 0, 0, datetime.timezone.utc)
+
+        constraintList = [Clusters.DeviceEnergyManagement.Structs.ConstraintsStruct(startTime=now.total_seconds() + 10, duration=20, nominalPower=forecast.slots[0].nominalPower, maximumEnergy=forecast.slots[0].nominalEnergy),
+                          Clusters.DeviceEnergyManagement.Structs.ConstraintsStruct(startTime=now.total_seconds(
+                          ) + 50, duration=20, nominalPower=forecast.slots[0].nominalPower, maximumEnergy=forecast.slots[0].nominalEnergy),
+                          Clusters.DeviceEnergyManagement.Structs.ConstraintsStruct(startTime=now.total_seconds() + 30, duration=20, nominalPower=forecast.slots[0].nominalPower, maximumEnergy=forecast.slots[0].nominalEnergy)]
+        await self.send_request_constraint_based_forecast(constraintList, cause=Clusters.DeviceEnergyManagement.Enums.AdjustmentCauseEnum.kLocalOptimization, expected_status=Status.ConstraintError)
+
+        self.step("9")
+        absMaxPower = await self.read_dem_attribute_expect_success(attribute="AbsMaxPower")
+
+        self.step("9a")
+        constraintList = [Clusters.DeviceEnergyManagement.Structs.ConstraintsStruct(
+            startTime=forecast.startTime, duration=forecast.slots[0].defaultDuration, nominalPower=absMaxPower + 1, maximumEnergy=forecast.slots[0].nominalEnergy)]
+        await self.send_request_constraint_based_forecast(constraintList, cause=Clusters.DeviceEnergyManagement.Enums.AdjustmentCauseEnum.kLocalOptimization, expected_status=Status.ConstraintError)
+
+        self.step("10")
+        absMinPower = await self.read_dem_attribute_expect_success(attribute="AbsMinPower")
+
+        self.step("10a")
+        constraintList = [Clusters.DeviceEnergyManagement.Structs.ConstraintsStruct(
+            startTime=forecast.startTime, duration=forecast.slots[0].defaultDuration, nominalPower=absMinPower - 1, maximumEnergy=forecast.slots[0].nominalEnergy)]
+        await self.send_request_constraint_based_forecast(constraintList, cause=Clusters.DeviceEnergyManagement.Enums.AdjustmentCauseEnum.kLocalOptimization, expected_status=Status.ConstraintError)
+
+        self.step("11")
+        constraintList = [Clusters.DeviceEnergyManagement.Structs.ConstraintsStruct(
+            startTime=forecast.startTime, duration=forecast.slots[0].defaultDuration, nominalPower=forecast.slots[0].nominalPower)]
+        await self.send_request_constraint_based_forecast(constraintList, cause=Clusters.DeviceEnergyManagement.Enums.AdjustmentCauseEnum.kLocalOptimization, expected_status=Status.InvalidCommand)
```

Comment:
I'm not sure why this should result in an invalid command? Is it because it isn't changing anything?
(maybe add a comment as to why this results in Invalid command?)

Reply:
The nominal energy was not specified